src/DocumentFormat.OpenXml/ParaId/Paragraph.ParaId.cs

```diff
@@ -0,0 +1,128 @@
+                }
+
+                return _wordprocessingDocument;
+            }
+        }
+
+        /// <summary>
+        /// Sets named property's value, registering w14:paraId values.
+        /// </summary>
+        /// <typeparam name="TSimpleType">The property type.</typeparam>
+        /// <param name="value">The property value.</param>
+        /// <param name="propertyName">The property name.</param>
+        private protected override void SetAttribute<TSimpleType>(TSimpleType? value, [CallerMemberName] string propertyName = null!)
+            where TSimpleType : class
+        {
+            if (propertyName is nameof(ParagraphId) && value is HexBinaryValue hexBinaryValue)
+            {
+                // TODO: Discuss how we want to deal with duplicate w14:paraId values assigned by the caller.
+                // The RegisterParagraphId() method returns true if the hexBinaryValue was unique and false otherwise.
+                // Therefore, if desired (e.g., always or based on settings), we could throw an exception.
+                WordprocessingDocument?.RegisterParagraphId(hexBinaryValue);
+            }
+
+            base.SetAttribute(value, propertyName);
+        }
+
+        /// <summary>
+        /// Attempts to set the attribute to a known attribute and registers any existing
+        /// w14:paraId values with the <see cref="WordprocessingDocument" /> instance.
+        /// </summary>
+        /// <remarks>
+        /// This is the Paragraph-specific implementation that registers the w14:paraId
+        /// values when loading the DOM tree.
+        /// </remarks>
+        /// <param name="qname">The attribute's qualified name.</param>
+        /// <param name="value">The attribute's value.</param>
+        /// <param name="strictRelationshipFound"></param>
+        /// <returns>true if the attribute is a known attribute.</returns>
+        private protected override bool TrySetFixedAttribute(in OpenXmlQualifiedName qname, string? value, bool strictRelationshipFound)
+        {
+            if (RawState.Attributes.Any())
+            {
+                if (strictRelationshipFound)
+                {
+                    return StrictTranslateAttribute(qname, value);
+                }
+
+                AttributeCollection.AttributeEntry attribute = RawState.Attributes[qname];
+
+                if (!attribute.IsNil)
+                {
+                    attribute.Value ??= attribute.Property.CreateNew();
+                    attribute.Value.InnerText = value;
+
+                    if (qname.Name == "paraId" && value is not null)
+                    {
+                        WordprocessingDocument?.RegisterParagraphId(value);
+                    }
+
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <inheritdoc />
+        public string SetUniqueParagraphId()
```

Comment:
Can we change this to a `bool TrySetUniqueId...` pattern rather than throw inside?

Reply:
Yes, that would be possible.